### PR TITLE
Mail.Ru oauth

### PR DESCRIPTION
**Pull request by gmist**
Added oauth for [Mail.Ru](https://en.wikipedia.org/wiki/Mail.Ru)

### CODE REVIEW

File: main/auth/mailru.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    'method': 'users.getInfo',
+    'app_id': mailru.consumer_key,
+    'session_key': access_token,
+    'secure': '1'
+  }
+  data['sig'] = mailru_sig(data)
+  me = mailru.get(
+    '/platform/api',
+    data=data
+  )
```
Comment (lipis) [inline main/auth/mailru.py]: one line maybe?

File: main/auth/mailru.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    flask.flash(u'You denied the request to sign in.')
+    return flask.redirect(util.get_next_url())
+
+  access_token = response['access_token']
+  flask.session['oauth_token'] = (access_token, '')
+  data = {
+    'method': 'users.getInfo',
+    'app_id': mailru.consumer_key,
+    'session_key': access_token,
+    'secure': '1'
```
Comment (lipis) [inline main/auth/mailru.py]: daggling comma..

File: main/auth/mailru.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+import auth
+import config
+import model
+import util
+
+from main import app
+
+mailru_config = dict(
+  # access_token_params={'grant_type': 'authorization_code'},
```
Comment (lipis) [inline main/auth/mailru.py]: ?

File: main/auth/mailru.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  base_url='https://www.appsmail.ru/',
+  consumer_key=config.CONFIG_DB.mailru_app_id,
+  consumer_secret=config.CONFIG_DB.mailru_app_secret,
+)
+
+mailru = auth.create_oauth_app(mailru_config, 'mailru')
+
+
+def mailru_sig(data):
+  param_list = sorted(list(item + '=' + data[item] for item in data))
```
Comment (lipis) [inline main/auth/mailru.py]: maybe `'%s=%s' % (item, data[item])` as usually?

**Comment by gmist** [comment]
Any other comments or suggestions?

**Comment by lipis** [comment]
I thought I merged it already.. totally forgot.. :) :beer:

**Pull request merged**